c-dependencies/js-compute-runtime/builtins/backend.cpp

```diff
@@ -0,0 +1,705 @@
+
+  JS::RootedValue name_val(cx);
+  if (!JS_GetProperty(cx, configuration, "name", &name_val)) {
+    return false;
+  }
+  if (!Backend::set_name(cx, backend, name_val)) {
+    return false;
+  }
+
+  JS::RootedValue target_val(cx);
+  if (!JS_GetProperty(cx, configuration, "target", &target_val)) {
+    return false;
+  }
+  if (!Backend::set_target(cx, backend, target_val)) {
+    return false;
+  }
+
+  bool found;
+  JS::RootedValue hostOverride_val(cx);
+  if (!JS_HasProperty(cx, configuration, "hostOverride", &found)) {
+    return false;
+  }
+  if (found) {
+    if (!JS_GetProperty(cx, configuration, "hostOverride", &hostOverride_val)) {
+      return false;
+    }
+    if (!Backend::set_host_override(cx, backend, hostOverride_val)) {
+      return false;
+    }
+  }
+
+  /// Timeouts for backends must be less than 2^32 milliseconds, or
+  /// about a month and a half.
+  JS::RootedValue connectTimeout_val(cx);
+  if (!JS_HasProperty(cx, configuration, "connectTimeout", &found)) {
+    return false;
+  }
+  if (found) {
+    if (!JS_GetProperty(cx, configuration, "connectTimeout", &connectTimeout_val)) {
+      return false;
+    }
+    double connectTimeout;
+    if (!JS::ToNumber(cx, connectTimeout_val, &connectTimeout)) {
+      return false;
+    }
+    int64_t connectTimeoutInt = std::round(connectTimeout);
+    if (connectTimeoutInt < 0) {
+      JS_ReportErrorNumberASCII(cx, GetErrorMessage, nullptr,
+                                JSMSG_BACKEND_CONNECT_TIMEOUT_NEGATIVE);
+      return false;
+    }
+    if (connectTimeoutInt >= std::pow(2, 32)) {
+      JS_ReportErrorNumberASCII(cx, GetErrorMessage, nullptr,
+                                JSMSG_BACKEND_CONNECT_TIMEOUT_TOO_BIG);
+      return false;
+    }
+    JS::SetReservedSlot(backend, Backend::Slots::ConnectTimeout, JS::Int32Value(connectTimeoutInt));
+  }
+
+  /// Timeouts for backends must be less than 2^32 milliseconds, or
+  /// about a month and a half.
+  JS::RootedValue firstByteTimeout_val(cx);
+  if (!JS_HasProperty(cx, configuration, "firstByteTimeout", &found)) {
+    return false;
+  }
+  if (found) {
+    if (!JS_GetProperty(cx, configuration, "firstByteTimeout", &firstByteTimeout_val)) {
+      return false;
+    }
+    double firstByteTimeout;
+    if (!JS::ToNumber(cx, firstByteTimeout_val, &firstByteTimeout)) {
+      return false;
+    }
+    int64_t firstByteTimeoutInt = std::round(firstByteTimeout);
+    if (firstByteTimeoutInt < 0) {
+      JS_ReportErrorNumberASCII(cx, GetErrorMessage, nullptr,
+                                JSMSG_BACKEND_FIRST_BYTE_TIMEOUT_NEGATIVE);
+      return false;
+    }
+    if (firstByteTimeoutInt >= std::pow(2, 32)) {
+      JS_ReportErrorNumberASCII(cx, GetErrorMessage, nullptr,
+                                JSMSG_BACKEND_FIRST_BYTE_TIMEOUT_TOO_BIG);
+      return false;
+    }
+    JS::SetReservedSlot(backend, Backend::Slots::FirstByteTimeout,
+                        JS::Int32Value(firstByteTimeoutInt));
+  }
```

Comment:
Would it be possible to extract this logic to a function? it looks pretty similar to the cases for `betweenBytesTimeout` and `connectTimeout`.

Reply:
good idea 👍 This was refactored in [`4d0249f` (#250)](https://github.com/fastly/js-compute-runtime/pull/250/commits/4d0249f36f8b5b9103dd87161c019a201edd76d0)